### PR TITLE
added feature to select optgroup and toggle options under optgroup

### PR DESCRIPTION
**Pull request by meetmagdalene**
If a user applies an id to an optgroup and that same optgroup id as a class of the options inside that optgroup, when a user clicks the optgroup label it toggles selecting the options in that optgroup.

### CODE REVIEW

**Comment by Khrysller** [comment]
Can I see this feature in a jsfiddle, @meetmagdalene? Thanks.

**Comment by meetmagdalene** [comment]
Will try and make one up tomorrow. I've never used JSFiddle before, so, more new things this month. :)

File: js/bootstrap-select.js [hunk 1]
```diff
@@ -409,11 +399,12 @@
        * @param [optgroup]
        * @returns {string}
        */
-      var generateLI = function (content, index, classes, optgroup) {
+      var generateLI = function (content, index, classes, id, optgroup) {
```
Comment (meetmagdalene) [inline js/bootstrap-select.js]: this is where you can add the optgroup label as a li id for the .dropdown-header

File: js/bootstrap-select.js [hunk 2]
```diff
@@ -470,6 +455,7 @@
             optID += 1;
 
             // Get the opt group label
+            var optgroupLabel = $this.parent().attr('label').replace(/\s+/g, '-').toLowerCase();
```
Comment (meetmagdalene) [inline js/bootstrap-select.js]: Grab the label from the optgroup

File: js/bootstrap-select.js [hunk 3]
```diff
@@ -479,7 +465,7 @@
               _li.push(generateLI('', null, 'divider', optID + 'div'));
             }
 
-            _li.push(generateLI(label, null, 'dropdown-header', optID));
+            _li.push(generateLI(label, null, 'dropdown-header', optgroupLabel, optID));
```
Comment (meetmagdalene) [inline js/bootstrap-select.js]: set the optgroup label as the id for this .dropdown-header

**Comment by caseyjhol** [comment]
Thanks for the pull request. Unfortunately this has a few issues, namely using an old version of bootstrap-select and changing numerous default options. Thanks for contributing, though! I was working on implementing this, but ran into a wall when trying to decide how to handle optgroups with maxOptions set. See https://github.com/silviomoreto/bootstrap-select/issues/737#issuecomment-97983484 for more information.

**Pull request closed without merging**